### ISSUE TITLE
On Premise TFS 2015 update 3. Installed Print task button no showing to print work item on backlog or any other place

### ISSUE THREAD

**Issue by kriskoneru** [MrTarantula/WIPrint#3]
we have on premise TFS 2015 I installed task on collection. but still the print button is not coming up on the backlog as well as on the queries.

![capture-1](https://user-images.githubusercontent.com/16949860/27799348-d11dfbb2-5fe2-11e7-9f08-181f65597601.PNG)
![capture-2](https://user-images.githubusercontent.com/16949860/27799374-e7ec9f4c-5fe2-11e7-9e3e-9ff88c91612b.PNG)

**Comment by MrTarantula** [comment]
Did you make sure it is installed on the project, not only the collection? If you're sure it's installed on the project, open developer tools with F12 and copy/paste the console log into a comment here.

**Comment by kriskoneru** [comment]
in on premise TFS.. the extensions can only be installed on Collection level not on project..
I have installed previously task on collection level only.

![capture -3](https://user-images.githubusercontent.com/16949860/27800494-9c86f034-5fe7-11e7-9f2e-f679fd1fea40.PNG)

**Comment by MrTarantula** [comment]
Ok, I was wrong about that. I'm not in front of a computer at the moment,  and I couldn't remember if there was a project-level installation step. A console log would help me narrow down the problem. I don't have my own instance of TFS 2015 to develop against.

**Comment by kriskoneru** [comment]
I did not understand what do you mean by console log.. I did check TFS logs on the server. there  is nothing related to the task that is being installed.

**Comment by kriskoneru** [comment]
Nothing no errors.. it says the two warnings.. 

but when I checked on the collection it says the task has already been installed.


![capture- 6](https://user-images.githubusercontent.com/16949860/27804443-f2e3d96e-5ffc-11e7-9174-fb0cd44b8872.PNG)
![capture-5](https://user-images.githubusercontent.com/16949860/27804442-f2e0d750-5ffc-11e7-943d-51564afa276e.PNG)

**Comment by MrTarantula** [comment]
It looks like you're using Internet Explorer. Can you try it in Chrome, Firefox, or Edge? If it's just Internet Explorer that will help me be able to debug it myself.

**Comment by MrTarantula** [comment]
I've tested in Internet Explorer in VSTS and it works. I think I know what the problem is though. I'm not near my dev environment but in the next few days I will have a test build for you to try.

**Comment by kriskoneru** [comment]
Hello, any update on the issue.. please let me know.

**Comment by MrTarantula** [comment]
I've created a test build. You can download it [here](https://github.com/MrTarantula/WIPrint/releases/download/1.0.4-test/mrtarantula.wiprint-1.0.4.vsix).

**Comment by kriskoneru** [comment]
Hello, thank you for making the changes..

but still the task is not showing up in the backlogs as well as the queries.

I am using TFS 2015 update 3 on Premise.


![capture -7](https://user-images.githubusercontent.com/16949860/27918374-c713b600-623c-11e7-96bd-daa122b858c7.PNG)
![capture -8](https://user-images.githubusercontent.com/16949860/27918376-c98f0632-623c-11e7-814f-959e9bb5b5a7.PNG)

**Comment by MrTarantula** [comment]
I'm not sure what could be causing the problem. It looks like you have other extensions installed. Can you try temporarily disabling any other extensions and see if it works?

**Comment by kriskoneru** [comment]
I disabled and removed all other extensions other than the print task. still it is not showing up.

did you test on TFS 2015 ??

**Comment by MrTarantula** [comment]
I began development of this extension on TFS 2015 but switched to TFS 2017. The extension points should be exactly the same. It's a very small script that doesn't use any 2017-specific features. I'm still not sure why it isn't working for you. Try this build [here](https://github.com/MrTarantula/WIPrint/releases/download/1.0.5-test/mrtarantula.wiprint-1.0.5.vsix).

**Comment by kriskoneru** [comment]
Hello, sorry to say that it is not working for TFS 2015 update 3 on premise..

The issue is all other tasks are working... like the Visualization task.

Does this task needs any dependency ???

**Comment by MrTarantula** [comment]
No dependencies required. It should just work. I asked a question earlier and never got a response: 

Have you tried it in Firefox, Chrome, or Edge? And do any errors show in those browsers (F12 works to show developer tools in all of those browsers).

Have you also tried completely uninstalling the extension and reinstalling? I don't know why it would fail to install, but it's worth a shot.

**Comment by kriskoneru** [comment]
yes, I tried in firefox, chrome but I did not try on Edge.. we are using windows 7 at our client location. 

yes, I completely uninstalled an installed the extension again.. ( The issue is it is not failing to install) It says it installed but the print button is not showing up in the backlog as well as the queries section.

I do not see any errors on the browsers.


![capture -9](https://user-images.githubusercontent.com/16949860/27960728-4fb45d30-62fa-11e7-9adb-713e28069187.PNG)

**Comment by kriskoneru** [comment]
Hello, we also have test tfs 2017 on premise .. I installed the task on tfs 2017 on premise it installed fine. but it doesnot work on tfs 2015 on premise.

**Comment by MrTarantula** [comment]
That's good to know. It must be a difference between the two API's. Without being able to debug on TFS 2015 it will be difficult for me to pin down where it fails. I've noticed that some errors in TFS don't bubble up to the console. I'll reach out to the TFS SDK developers to see what they have to say about it.

I include the sourcemap in my extension, so if you have a web developer who is familiar with debugging JavaScript or TypeScript they should be able to set breakpoints and find where it's failing. [Here](https://scotch.io/tutorials/debugging-javascript-with-chrome-devtools-breakpoints) is a guide to debugging in Chrome.

**Comment by MrTarantula** [comment]
OK. I was able to get TFS 2015 installed locally. I believe I have found the bug but it solves the problem in TFS 2015 while creating a problem for later versions. Until I have that figured out I can't release it through the Marketplace but you can download it [here](https://github.com/MrTarantula/WIPrint/releases/download/1.0.6-test/mrtarantula.wiprint-1.0.6.vsix).

**Comment by MrTarantula** [comment]
The first image has nothing to do with my extension. You cannot print an entire backlog. Going to `File > Print` or `Ctrl+P` does not go through my extension. You can select several (or all) work items with `Ctrl+Click` or `Shift+Click`, then right click and select `Print Selection`.

The query issue is a problem with Internet Explorer. It works in Chrome, but I spotted another bug. With [this update](https://github.com/MrTarantula/WIPrint/releases/download/1.0.7-test/mrtarantula.wiprint-1.0.7.vsix) you should be able to print from a query in Chrome. I will have to do some investigation to get it working in IE.

**Comment by kriskoneru** [comment]
Hello , I am able to print but I am custom fields on work item.. so the print does not print the custom work item fields..

**Comment by jeffyoungstrom** [comment]
I've got 2015 update 3 too. Loading your 1.0.6 version linked above presents the Print button, but printing fails. The Console shows this:

> :8080/tfs/ADV_RnD/_apis/wit/workItems?ids=214&fields=System.ID%2CSystem.Tit…osoft.VSTS.Scheduling.StartDate%2CMicrosoft.VSTS.Common.AcceptanceCriteria Failed to load resource: the server responded with a status of 400 (Bad Request)
> corejs?v=ITI0eYHCCehqqjk6JTyPaR5R793ETLoY2_Lqz191Pyo1:166 Rejected XDM promise with no reject callbacks
> n._reject @ corejs?v=ITI0eYHCCehqqjk6JTyPaR5R793ETLoY2_Lqz191Pyo1:166
> corejs?v=ITI0eYHCCehqqjk6JTyPaR5R793ETLoY2_Lqz191Pyo1:166 Object
> n._reject @ corejs?v=ITI0eYHCCehqqjk6JTyPaR5R793ETLoY2_Lqz191Pyo1:166

Loading the URL in the first message gives this output:
> {"$id":"1","innerException":null,"message":"TF51535: Missing or unsupported field name Microsoft.VSTS.Common.AcceptanceCriteria.","typeName":"Microsoft.TeamFoundation.WorkItemTracking.Server.Metadata.WorkItemTrackingFieldDefinitionNotFoundException, Microsoft.TeamFoundation.WorkItemTracking.Server, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","typeKey":"WorkItemTrackingFieldDefinitionNotFoundException","errorCode":0,"eventId":3200}`

This is from a team project using the Agile template. In another team project using Scrum I instead get a failure on DueDate:

> {"$id":"1","innerException":null,"message":"TF51535: Missing or unsupported field name Microsoft.VSTS.Scheduling.DueDate.","typeName":"Microsoft.TeamFoundation.WorkItemTracking.Server.Metadata.WorkItemTrackingFieldDefinitionNotFoundException, Microsoft.TeamFoundation.WorkItemTracking.Server, Version=14.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","typeKey":"WorkItemTrackingFieldDefinitionNotFoundException","errorCode":0,"eventId":3200}

**Comment by MrTarantula** [comment]
The problem, which you have probably figured out, is that right now the work item fields are hardcoded, and if any single field doesn't exist for the project type it's an immediate 400 bad request for the query. I plan on remedying this when I have some time, but it will take a bit of work.

**Comment by MrTarantula** [comment]
@kriskoneru @jeffyoungstrom I have a new [test build](https://github.com/MrTarantula/WIPrint/releases/download/1.0.8-test/mrtarantula.wiprint-1.0.8.vsix) that should allow printing from any work item using any template, including custom fields.

I still have not found a solution to the other issues.

**Comment by jeffyoungstrom** [comment]
I'm still getting the same errors as in my earlier post using your 1.0.8 test build. But maybe that's the "other issues" you're talking about...

**Comment by kriskoneru** [comment]
I am not able to print custom fields. I installed new version of the task.

**Comment by MrTarantula** [comment]
I am in the process of adding more customization and would like some feedback. You an download a test extension [here for 2017/VSTS](https://github.com/MrTarantula/WIPrint/releases/download/v1.1.1-test/mrtarantula.wiprint-dev-1.1.1.vsix)  and [here for 2015](https://github.com/MrTarantula/WIPrint/releases/download/v1.1.2-2015test/mrtarantula.wiprint-dev-1.1.2.vsix) to try it out. To customize fields you will need to go to your project's admin settings. There will be a tab called "Work Item Print" where you can select all the fields you want.

You can install it side-by-side with the other extension, the menu items are prefixed with "DEV".

**NOTE: I have not figured out the problems with printing in IE yet.**

**Comment by jeffyoungstrom** [comment]
Where might I find "project's admin settings". I've been clicking around for half an hour and I'm not seeing it. Thanks.

**Comment by MrTarantula** [comment]
It's the gear icon. When in a project, click the gear icon. Then there will be a tab (in Microsoft's terms a hub) called "Work Item Print".

VSTS/2017:
![image](https://user-images.githubusercontent.com/11617455/28438575-3e3b04e0-6d64-11e7-8e66-c035e8a64f63.png)

2015:
![image](https://user-images.githubusercontent.com/11617455/28438856-30a2959a-6d65-11e7-8aa5-d5129a38b0a2.png)

**Comment by jeffyoungstrom** [comment]
Ah, thanks. I was looking there, but it turned out I'd uploaded the new build but hadn't installed it. Doh.

Basically works. Yay! (again, I'm using TFS 2015.3 in Chrome)

Be nice if the list of all fields was sorted. And if the ordering of the selected fields could be adjusted. And maybe the standard field picker behavior that things on the right side don't show up in the list on the left side.

Not too surprised that when I add History to the list it prints just the word "undefined" Be nice if that could work. Similarly if it could do something reasonable for other collection fields like "Related Links" and "Attached Files"

Tags works if there's only one, but shows undefined if there are multiples.

Might be nice to have an option to exclude any fields that would show "undefined". I added all the fields for our Bug type, and most of them are undefined for a Task. Would be good to have as an option so can tell the difference between a field being missing for other reasons vs. undefined...

I'll play around with it some more. Is there a better way to pass along feedback than this thread?

**Comment by jeffyoungstrom** [comment]
Oh, if you want to make me really happy, add the ability to print all the work items returned by a query in one go.

**Comment by MrTarantula** [comment]
Testing with VSTS it was automatically sorted. It was only after uploading the build that I thought to test 2015. Sorting can be fixed easily enough.

I plan on implementing all of the usual field picker behaviors, just wanted to get something working and get feedback.

The other issues I'll have to do some runtime type-checking to display properly. Also, I wasn't sure whether to hide fields without value or display a message.

This is the best place to comment for issues. If you'd like you can open your own issue to track your problems.

**Comment by jeffyoungstrom** [comment]
I don't know if there's anything you can do about this, but the print preview window is blocking for other TFS tabs in the same window on chrome. i.e., if I print a work item and don't push the print button on the preview and switch to another tab in the same window, TFS actions in that window don't respond properly (and it's not immediately obvious why).

**Comment by MrTarantula** [comment]
> Oh, if you want to make me really happy, add the ability to print all the work items returned by a query in one go.

The 2017 version has a button at the top of the query results for just that purpose. I'll have to look to see why it's not showing up in 2015. It can handle ~300 work items at once.

**Comment by kriskoneru** [comment]
hello , thank you very much the print is now working and I am able to print the custom fields.

But yeah it shows all the fields and we have so many custom fields in so many projects and all of them show up along with Out of box fields.

As the other user mentioned if you can help in creating a query that and sorting it out that would be a lot helpful.

and eagerly waiting for your update to work on IE.

but thank you very much on working on this.

**Comment by kriskoneru** [comment]
wow this new extension allows me to print even from IE from backlog. But I can not print from queries. the extension does not work.

One more request how do I change the order of printing on the output because it prints on same order in the output that I selected and if I want to change the order it does not allow me to do that. either I have to remove everything and again then choose one by one.

**Comment by MrTarantula** [comment]
Sorry, there is no way to change the order. Well, it may be *possible*, but it would greatly increase the time it takes to print, especially with more items. It's not something I plan to add. Sorry.

I've compiled a list of issues into #6 to track them all. Let me know there if I am missing anything. I will have a new test build available later today addressing some of those problems.

**Comment by kriskoneru** [comment]
No issue, I am fine with it.

**Comment by MrTarantula** [comment]
New Test build [here](https://github.com/MrTarantula/WIPrint/releases/download/v1.1.3-test/mrtarantula.wiprint-dev-1.1.3.vsix). It should work for both TFS 2015 and 2017/VSTS, but I can't test on 2015 until Monday. Please test the fieldpicker thoroughly and let me know if it's behaving properly. I think I fixed printing in IE11 as well.

**Comment by kriskoneru** [comment]
using new build. the list is sorted and I can move the output print fields up and down. 

The issue is the PRINT itself is not working. can you please look into this issue.

**Comment by MrTarantula** [comment]
I think I squashed the bug you're talking about. Before, you could print from right-clicking an item or a selection of items, but printing from inside the item failed. [Here's another build](https://github.com/MrTarantula/WIPrint/releases/download/v1.1.4-test/mrtarantula.wiprint-dev-1.1.4.vsix).

**Comment by kriskoneru** [comment]
thank you very much. it is working now even on IE 11.

**Comment by MrTarantula** [comment]
That's great! In the next few days I will release a new official extension version with some new features like per work item type field customization. Once it's ready I will notify you here and close this issue.

**Comment by jeffyoungstrom** [comment]
Agreed, 1.1.4 looks good. I'm able to print multiples. The field picker is more friendly. Having the undefineds disappear is swell.

Looks like the History item just shows the text of the last entry in the History. I'd expect History to show the text of each change along with at least date/time and identity of the person making the change. Could maybe restrict to just the discussion entries like with the web UI "Discussion Only" option.

**Comment by kriskoneru** [comment]
Hello @MrTarantula I have urgent issue.. 
for stakeholders the settings icon does not exist so how can we get around this issue. can you please provide a workaround for this problem.

Any help would be greatly appreciated

![sample -1](https://user-images.githubusercontent.com/16949860/29171175-995da288-7da9-11e7-9af0-e0381b9c305e.jpg)

**Comment by MrTarantula** [comment]
No, that is not possible. There is no extension point for that menu. I put it in the admin settings because the page shouldn't need to be accessed very often, just to configure the fields. I can move it somewhere else, but I'd prefer it to be out of the way.

I'm making progress on a new version of the extension. I should have a test build available soon.

**Comment by kriskoneru** [comment]
@MrTarantula wow so can you change it to someplace else so that irrespective of the license the user has that they can configure the fields. this is important to our project. or can you give us an custom extension just for our project so that they can configure in the queries or some place. any help would be greatly appreciated.

**Comment by MrTarantula** [comment]
Here are two other places that should be visible to stakeholders. If I had to choose, I would rather it be a backlog tab than a hub. Do either of those make sense as a place to configure it?

![image](https://user-images.githubusercontent.com/11617455/29174243-4d09d36e-7dab-11e7-9ef7-fdf436034b36.png)

**Comment by MrTarantula** [comment]
Also, should the configuration be per user, or per project? My initial thought was that an administrator would set the fields they want, and everyone would have to abide by that configuration, which is why I put it in the admin settings. But if each user wants their own fields, then taking out of admin settings makes sense, but everyone will have to configure their own settings.

**Comment by kriskoneru** [comment]
Yes, backlog tab would be fine.  you can place the tab there.

The only question I have is still the work item print tab will be shown in the settings or not ??

If you can leave the existing option as it is that will be good and for the stake holders if you can add it at the backlog that would be awesome. 

I really appreciate for all the quick response. You are awesome.

**Comment by kriskoneru** [comment]
@MrTarantula but the current print button is not working per project. If it would be per project that would be great because for stakeholders they will just print. but it is not working in the way that you are thinking. It is behaving on individual basis. that is why the issue has been raised in the first place.  (it is asking everyone to select the settings) otherwise the print button is not working.

**Comment by MrTarantula** [comment]
OK. If it is per project, and placed as a backlog tab, then anyone will be able to modify the configuration. Is that acceptable?

I could make the tab only display for certain user groups, and make those groups configurable by an admin in the admin settings, but that would be much more work.

**Comment by kriskoneru** [comment]
@MrTarantula then it will cause issue. if some developer add/removes a field it will cause confusion to the business users(all of a sudden they will see the field not being printed) . I do not want this to happen. I think it would be fine on individual basis. They will select what fields they want to display on the print. So just add the tab on the backlog page. that would solve the issue.

**Comment by MrTarantula** [comment]
Bad news. In TFS 2015 there is no extension point for backlog tab, or if there was, I can't find any documentation for it. It will show up in the Queries, but not Backlog. So annoying. Here are all the [extension points] (https://www.visualstudio.com/en-us/docs/integrate/extensions/reference/targets/overview) currently available, but some of them may not be available in 2015. Is there anywhere else that it would make sense to put this?

**Comment by kriskoneru** [comment]
ok, can you put on the hub as show in your screen shot above. I am ok with it. at least it will solve the issue.

**Comment by kriskoneru** [comment]
@MrTarantula any update on the task. This is very useful task for our business users. If the extension point cannot be done at hub on work item tab. can you at least make the change so that administrators can select the columns so that all other people can print ???

can you please let me know.

**Comment by MrTarantula** [comment]
I have a new test build that offers per item customization, a better UI, and shows as a Work hub. You can download it [here](https://github.com/MrTarantula/WIPrint/releases/download/v1.2.0/mrtarantula.wiprint-dev-1.2.0.vsix).

I am encountering an issue in TFS 2015, though, and I'm not sure if it is my configuration or not. When I click on a work item type it does not load, and gives me a 404 error. Can you please test this? It works fine on VSTS.

**Comment by MrTarantula** [comment]
Please test it thoroughly if you can. All of the functionality was built from scratch. If the field picker functions differently than expected or is not intuitive then I want to know.

**Comment by kriskoneru** [comment]
I have uploaded the new task. I am able to see the work item print. but the configuration is not being saved one I select the columns. ( so I can not print all the fields )

One more request is I like instead of selecting work items and selecting columns per work item.

 Your old way is just select the columns irrespective of the work items.  can you make the change ??

**Comment by MrTarantula** [comment]
One thing different from the previous test builds is that you must click **Save** before it will save the fields after a change. Can you open your developer tools(F12) go to the Network tab, then click **Save**. If there's an error it should show there.

I am planning on creating a global config where you can save options like the date format and other things as I think of them. I could make it possible for a global work item configuration instead of a per work item type. All of this is way down on my list of priorities though.

**Comment by kriskoneru** [comment]
yes, when I click on save button the save criteria goes away and says no saved data.

I see access denied error on the console 
![capture -12](https://user-images.githubusercontent.com/16949860/29222089-191830fc-7e8f-11e7-9115-27f66df5dbbd.PNG)

**Comment by MrTarantula** [comment]
Can you find the error in the Network tab?

**Comment by kriskoneru** [comment]
No. No error.

**Comment by MrTarantula** [comment]
It looks like in IE you have to press a **Play** button on the Network tab before it will start showing up. I don't normally debug in IE.

![image](https://user-images.githubusercontent.com/11617455/29223003-9535d57c-7e8a-11e7-8f26-865cc0aedc92.png)

I haven't been able to replicate the issue.

**Comment by kriskoneru** [comment]
I did start and stop after the issue. 

I am sending the attachment. Please let me know.
[NetworkData.zip](https://github.com/MrTarantula/WIPrint/files/1218524/NetworkData.zip)

**Comment by MrTarantula** [comment]
That looks like the same error I was getting. Here's what you should do:

1. Delete the `Dev Work Item Print` extension
2. Clear browser cache
3. Close all open browser Windows
4. Reinstall `Dev Work Item Print` extension

That's what fixed it for me.

**Comment by MrTarantula** [comment]
I think the problem was caused when updating the extension over and over again. Not sure the exact reason.

**Comment by kriskoneru** [comment]
@MrTarantula nope not working. I cleared cache. I uninstall and removed. I reinstalled extension.

still the save is not working.

**Comment by MrTarantula** [comment]
Try uninstalling and then installing the extension from [here](https://github.com/MrTarantula/WIPrint/releases/download/v1.2.0-testagain/mrtarantula.wiprint-dev2-1.0.1.vsix).

**Comment by kriskoneru** [comment]
@MrTarantula it is working now. Thank you very much. :+1:

**Comment by MrTarantula** [comment]
@kriskoneru I have released a new version of the extension. You can find it in the [marketplace](https://marketplace.visualstudio.com/items?itemName=mrtarantula.wiprint). You can remove the development version of the extension now.

@jeffyoungstrom You should remove the development version as well. I will only be supporting the official version. You should find it at least as good as the development build, if not better.

If you like the extension, please rate or review it there. Thanks.

**Comment by jeffyoungstrom** [comment]
Thanks @MrTarantula, I'll snag the new version on Monday. Sounds like you've made some great improvements.

**Comment by kriskoneru** [comment]
@MrTarantula when users try to print using IE 11 while they are on backlog.  The print is working in such a way that it overlaps and shows the backlog and does not do a good job at printing. But this doesnot happen while they print from queries. 

Is there any way such that if the print redirect them to queries and then print from queries ??

![capture -11](https://user-images.githubusercontent.com/16949860/29779708-5d90c474-8be1-11e7-9d93-46ad3548defc.PNG)

**Comment by MrTarantula** [comment]
That's not my extension displaying all of that. Printing from the backlog works exactly the same as printing from from queries.  You can't go to `File > Print`. Select a work item or multiple items, then right-click them and select `Print` or `Print Selection`. Look at the [gifs on this page](https://github.com/MrTarantula/WIPrint/blob/master/details.md). If your users aren't doing one of those things, then they're not going through my extension.

**Comment by kriskoneru** [comment]
@MrTarantula here is what I am doing..  I am trying to print directly from the Menu. The screen shot that you put does not work for IE 11. You can try it by yourself as well. that is why I am asking if the button can redirect them to the queries so that print will be in a proper manner. 

Select  Print  from the work item form menu:

**Comment by MrTarantula** [comment]
OK, this is a legitimate bug. I was able to reproduce it in both TFS 2015 and VSTS. At the moment I'm not sure when I will be able to fix it, or even how to fix it without a lot of changes.

I'm not going to create a button to redirect to a query. That's outside the scope of what I'm trying to provide. The easiest way for them to print in IE is to back out of the item, then right-click on it and select `Print`.

**Comment by kriskoneru** [comment]
@MrTarantula yes, thank you. In the mean while  I will provide the users to do the same.

**Comment by kriskoneru** [comment]
@MrTarantula any update on the issue. are you able to find the a resolution in the time being ??

**Comment by MrTarantula** [comment]
No resolution yet. I haven't had time to look at it. I'm also going on vacation next week, so I wouldn't expect a fix for this for at least a few weeks.

**Comment by kriskoneru** [comment]
@MrTarantula any update on this issue. ??

Sorry for bugging you.

**Comment by kriskoneru** [comment]
@MrTarantula can you for the time being remove the print button on the work item itself.

**Comment by MrTarantula** [comment]
@kriskoneru With v2.0.5 I believe I have solved the last remaining problem with IE. Please let me know if it is working for you.

**Comment by kriskoneru** [comment]
@MrTarantula you are really really awesome. You saved me. Thank you very very much.

The print is now working for IE 11. 

I think this issue can be closed now. 

Thank you once again.

**Comment by MrTarantula** [comment]
If you have any other problems, please open a new issue.
